python_scripts/linear_models_sol_05.py

```diff
@@ -1,11 +1,12 @@
 # %% [markdown]
 # # 📃 Solution for Exercise M4.05
-# In the previous notebook, we presented a non-penalized logistic regression
-# classifier. This classifier accepts a parameter `penalty` to add a
-# regularization. The regularization strength is set using the parameter `C`.
-#
-# In this exercise, we ask you to train a l2-penalized logistic regression
-# classifier and to find by yourself the effect of the parameter `C`.
+# In the previous notebook we presented the `penalty` parameter of a logistic
+# regression classifier. It controls the **type** of regularization to use,
+# whereas the regularization **strength** is set using the parameter `C`. In
+# that case we set `penalty="none"`, which is equivalent to an infinitely large
```

Comment:
In the example, you are not setting the penalty, so by default, you have a `"l2"` penalty which is the gotcha here. I would still force to use a l2 penalty. We can be explicit that we want the same penalty than in Ridge that shrinks weight magnitude towards zero and that is called l2.

Reply:
Done, thanks!

Reply:
There was a misunderstanding, see my review below and my proposal to make the phrasing more explicit.